packages/contracts/hardhat.config.ts

```diff
@@ -9,6 +9,27 @@ import "@nomicfoundation/hardhat-verify";
 import "hardhat-docgen";
 import "@openzeppelin/hardhat-upgrades";
 import { solidityConfig } from "./hardhat.solidity-config";
+import { getDeployerWallet } from "./scripts/util/get-deployer-wallet";
+
+// ----- WALLET CONFIGURATION -----
+// initialize with dummy fallback private key
+let PRIVATE_KEY: string =
+  "DEAD000000000000000000000000000000000000000000000000000000000000";
+// if process argument "run" is present, use the deployer wallet
+// @dev use arg "run" check to avoid wallet nuisance when hardhat
+// runs pre-processes such as "compile" prior to running scripts
+if (process.argv.includes("run")) {
```

Comment:
🚀

Reply:
I realized this had a smol bug, but fixed using similar trick in [83325e4](https://github.com/ArtBlocks/artblocks-contracts/pull/1206/commits/83325e40a8b492c8602516217c1e4c4e15eb1db5)